File: sample/src/main/java/com/lazygeniouz/house/ads/sample/fragments/NativeAd.java
package com.lazygeniouz.house.ads.sample.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.SwitchCompat;

import com.google.android.material.button.MaterialButton;
import com.lazygeniouz.house.ads.HouseAdsNative;
import com.lazygeniouz.house.ads.listener.NativeAdListener;
import com.lazygeniouz.house.ads.sample.R;

public class NativeAd extends BaseFragment {
    public NativeAd() {}

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.native_ad, container, false);
    }

    @SuppressWarnings("ConstantConditions")
    @Override
    public void onViewCreated(@NonNull View rootView, @Nullable Bundle savedInstanceState) {
        boolean isShowLocalAssets = getContext().getSharedPreferences("localAssetsNative", Context.MODE_PRIVATE).getBoolean("value", false);

        rootView.findViewById(R.id.card_view).setVisibility(View.GONE);
        TextView loading = rootView.findViewById(R.id.loading);

        SwitchCompat localAssets = rootView.findViewById(R.id.useLocalResources);
        HouseAdsNative houseAdsNative;
        if (!isShowLocalAssets) {
            rootView.findViewById(R.id.houseAds_header_image).setVisibility(View.GONE);
            houseAdsNative = new HouseAdsNative(getContext(), "https://lz-houseads.firebaseapp.com/houseAds/ads.json");
        }
        else houseAdsNative = new HouseAdsNative(getContext(), R.raw.ad_assets);

        houseAdsNative.setNativeAdView(rootView.findViewById(R.id.card_view))
                .usePalette(true)
                .setNativeAdListener(new NativeAdListener() {
                    @Override
                    public void onAdLoaded() {
                        loading.setVisibility(View.GONE);
                        rootView.findViewById(R.id.card_view).setVisibility(View.VISIBLE);
                    }

                    @Override
                    public void onAdLoadFailed(@NonNull Exception e) {
                        loading.setText(String.format("%s%s", getString(R.string.ad_failed), e.getMessage()));
                        loading.setVisibility(View.VISIBLE);
                    }
                });

        MaterialButton load = rootView.findViewById(R.id.load);
        load.setOnClickListener(v -> {
            rootView.findViewById(R.id.card_view).setVisibility(View.GONE);
            loading.setVisibility(View.VISIBLE);
            houseAdsNative.loadAds();
        });

        localAssets.setChecked(getContext().getSharedPreferences("localAssetsNative", Context.MODE_PRIVATE).getBoolean("value", false));
        localAssets.setOnCheckedChangeListener((buttonView, isChecked) -> {
            getContext().getSharedPreferences("localAssetsNative", Context.MODE_PRIVATE).edit().putBoolean("value", isChecked).apply();
            getActivity().recreate();
        });

    }
}

File: sample/src/main/java/com/lazygeniouz/house/ads/sample/fragments/NativeAd.kt
package com.lazygeniouz.house.ads.sample.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.widget.SwitchCompat
import com.google.android.material.button.MaterialButton
import com.lazygeniouz.house.ads.HouseAdsNative
import com.lazygeniouz.house.ads.listener.NativeAdListener
import com.lazygeniouz.house.ads.sample.R

class NativeAd : BaseFragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.native_ad, container, false)
    }

    override fun onViewCreated(rootView: View, savedInstanceState: Bundle?) {
        val isShowLocalAssets = requireContext().getSharedPreferences("localAssetsNative", Context.MODE_PRIVATE).getBoolean("value", false)
        rootView.findViewById<View>(R.id.card_view).visibility = View.GONE
        val loading = rootView.findViewById<TextView>(R.id.loading)
        val localAssets: SwitchCompat = rootView.findViewById(R.id.useLocalResources)
        val houseAdsNative: HouseAdsNative
        if (!isShowLocalAssets) {
            rootView.findViewById<View>(R.id.houseAds_header_image).visibility = View.GONE
            houseAdsNative = HouseAdsNative(requireContext(), "https://lz-houseads.firebaseapp.com/houseAds/ads.json")
        } else houseAdsNative = HouseAdsNative(requireContext(), R.raw.ad_assets)
        houseAdsNative.setNativeAdView(rootView.findViewById(R.id.card_view))
                .usePalette(true)
                .setNativeAdListener(object : NativeAdListener {
                    override fun onAdLoaded() {
                        loading.visibility = View.GONE
                        rootView.findViewById<View>(R.id.card_view).visibility = View.VISIBLE
                    }

                    override fun onAdLoadFailed(exception: Exception) {
                        loading.text = String.format("%s%s", getString(R.string.ad_failed), exception.message)
                        loading.visibility = View.VISIBLE
                    }
                })
        val load: MaterialButton = rootView.findViewById(R.id.load)
        load.setOnClickListener {
            rootView.findViewById<View>(R.id.card_view).visibility = View.GONE
            loading.visibility = View.VISIBLE
            houseAdsNative.loadAds()
        }
        localAssets.isChecked = requireContext().getSharedPreferences("localAssetsNative", Context.MODE_PRIVATE).getBoolean("value", false)
        localAssets.setOnCheckedChangeListener { _, isChecked ->
            requireContext().getSharedPreferences("localAssetsNative", Context.MODE_PRIVATE).edit().putBoolean("value", isChecked).apply()
            requireActivity().recreate()
        }
    }
}